event/event.go

```diff
@@ -180,8 +184,12 @@ func parseAsCloudEventBinary(headers http.Header, payload interface{}) (*Event,
 		return nil, err
 	}
 
-	if val, err := time.Parse(time.RFC3339, headers.Get("CE-EventTime")); err == nil {
-		event.EventTime = val
+	if headers.Get("CE-EventTime") != "" {
+		if val, err := time.Parse(time.RFC3339, headers.Get("CE-EventTime")); err == nil {
```

Comment:
I would flip this around, the val is stlll local to the outer `if` block

```
if headers.Get("CE-EventTime") != "" {
    val, err := time.Parse(time.RFC3339, headers.Get("CE-EventTime"))
    if err != nil {
        return nil, err
    }
    event.EventTime = &val
}

Reply:
Oh, yeah. One sec